unified-runtime/source/adapters/cuda/common.cpp

```diff
@@ -129,3 +131,84 @@ void setPluginSpecificMessage(CUresult cu_res) {
   setErrorMessage(message, UR_RESULT_ERROR_ADAPTER_SPECIFIC);
   free(message);
 }
+
+namespace umf {
+
+ur_result_t getProviderNativeError(const char *, int32_t) {
+  // TODO: implement when UMF supports CUDA
+  return UR_RESULT_ERROR_UNKNOWN;
+}
+
+// Create UMF CUDA memory provider for the host memory (UMF_MEMORY_TYPE_HOST)
+// from a device
+ur_result_t
+createHostMemoryProvider(ur_device_handle_t_ *DeviceHandle,
```

Comment:
what's the reason for passing DeviceHandle here? Looks like you only need context handle.

Reply:
There is `CUcontext context = DeviceHandle->getNativeContext();` three lines below.

Reply:
Right, that's why I'm asking. Why not just pass CuContext here?s

Reply:
Right.
Done.